### ISSUE TITLE
hypergan: command not found

### ISSUE THREAD

**Issue by mcDandy** [HyperGAN/HyperGAN#115]
I did everything correctly, but hypergan command does not work

daniel@daniel-pc:~$ pip3 install hypergan --upgrade
Collecting hypergan
  Downloading hypergan-0.9.4.tar.gz
Building wheels for collected packages: hypergan
  Running setup.py bdist_wheel for hypergan ... done
  Stored in directory: /home/daniel/.cache/pip/wheels/e4/c8/6b/f5111f93fa7cd698aa9d67e99f04bcdda76f6cd5139a90f50f
Successfully built hypergan
Installing collected packages: hypergan
Successfully installed hypergan-0.9.4
daniel@daniel-pc:~$  pip3 install numpy tensorflow-gpu hyperchamber pillow pygame
Collecting numpy
  Downloading numpy-1.13.1-cp35-cp35m-manylinux1_x86_64.whl (16.9MB)
    100% |████████████████████████████████| 16.9MB 101kB/s 
Collecting tensorflow-gpu
  Downloading tensorflow_gpu-1.2.1-cp35-cp35m-manylinux1_x86_64.whl (89.2MB)
    100% |████████████████████████████████| 89.2MB 17kB/s 
Collecting hyperchamber
Collecting pillow
  Downloading Pillow-4.2.1-cp35-cp35m-manylinux1_x86_64.whl (5.8MB)
    100% |████████████████████████████████| 5.8MB 232kB/s 
Collecting pygame
  Using cached pygame-1.9.3-cp35-cp35m-manylinux1_x86_64.whl
Collecting markdown>=2.6.8 (from tensorflow-gpu)
  Downloading Markdown-2.6.8.tar.gz (307kB)
    100% |████████████████████████████████| 317kB 441kB/s 
Collecting six>=1.10.0 (from tensorflow-gpu)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting html5lib==0.9999999 (from tensorflow-gpu)
Collecting bleach==1.5.0 (from tensorflow-gpu)
  Using cached bleach-1.5.0-py2.py3-none-any.whl
Collecting protobuf>=3.2.0 (from tensorflow-gpu)
  Using cached protobuf-3.3.0-cp35-cp35m-manylinux1_x86_64.whl
Collecting werkzeug>=0.11.10 (from tensorflow-gpu)
  Using cached Werkzeug-0.12.2-py2.py3-none-any.whl
Collecting wheel>=0.26 (from tensorflow-gpu)
  Using cached wheel-0.29.0-py2.py3-none-any.whl
Collecting backports.weakref==1.0rc1 (from tensorflow-gpu)
  Using cached backports.weakref-1.0rc1-py3-none-any.whl
Collecting requests>=2.4.2 (from hyperchamber)
  Using cached requests-2.18.1-py2.py3-none-any.whl
Collecting olefile (from pillow)
Collecting setuptools (from protobuf>=3.2.0->tensorflow-gpu)
  Using cached setuptools-36.0.1-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.4.2->hyperchamber)
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.6,>=2.5 (from requests>=2.4.2->hyperchamber)
  Using cached idna-2.5-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.4.2->hyperchamber)
  Using cached certifi-2017.4.17-py2.py3-none-any.whl
Collecting urllib3<1.22,>=1.21.1 (from requests>=2.4.2->hyperchamber)
  Using cached urllib3-1.21.1-py2.py3-none-any.whl
Building wheels for collected packages: markdown
  Running setup.py bdist_wheel for markdown ... done
  Stored in directory: /home/daniel/.cache/pip/wheels/85/a7/08/33ee5cd488d0365d8bed79d1d4e5c28dd3fbfc7f6d0ad4bb09
Successfully built markdown
Installing collected packages: numpy, markdown, six, html5lib, bleach, setuptools, protobuf, werkzeug, wheel, backports.weakref, tensorflow-gpu, chardet, idna, certifi, urllib3, requests, hyperchamber, olefile, pillow, pygame
Successfully installed backports.weakref-1.0rc1 bleach-1.5.0 certifi-2017.4.17 chardet-3.0.4 html5lib-0.9999999 hyperchamber-0.2.8 idna-2.5 markdown-2.6.8 numpy-1.13.1 olefile-0.44 pillow-4.2.1 protobuf-3.3.0 pygame-1.9.3 requests-2.18.1 setuptools-36.0.1 six-1.10.0 tensorflow-gpu-1.2.1 urllib3-1.21.1 werkzeug-0.12.2 wheel-0.29.0

But every hypergan command I write shows this console output. hypergan: command not found
When I try to use virtualenv it doesent work eather
daniel@daniel-pc:~$ sudo virtualenv --system-site-packages -p python3 hypergan
Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/daniel/hypergan/bin/python3
Not overwriting existing python script /home/daniel/hypergan/bin/python (you must use /home/daniel/hypergan/bin/python3)
Installing setuptools, pkg_resources, pip, wheel...done.
daniel@daniel-pc:~$ hypergan
hypergan: příkaz nebyl nalezen
daniel@daniel-pc:~$   source hypergan/bin/activate
(hypergan) daniel@daniel-pc:~$ hypergan
hypergan: příkaz nebyl nalezen

**Comment by mcDandy** [comment]
Am I doing something wrong?

**Comment by martyn** [comment]
Hi,

If you run `pip3 show -f hypergan` it will show you the files that were installed.

```
Location: /usr/local/lib/python3.5/site-packages
...
Files:
  ../../../bin/hypergan
...
```

Then you can check for the file by joining the paths:

`ls -la /usr/local/lib/python3.5/site-packages/../../../bin/hypergan`

**Comment by mcDandy** [comment]
It appers that file //bin/hypergan does not exist

**Comment by mcDandy** [comment]
using ubuntu 17.04

**Comment by martyn** [comment]
Interesting.  I think it's probably a path issue.  Does `pip3 show -f hypergan` show anything?

**Comment by mcDandy** [comment]
yes. It shows a bunch of files.

**Comment by martyn** [comment]
Can you share the output?  I think it's probably a path issue

**Comment by mcDandy** [comment]
I will will when I boot ito ubuntu. I am currently in windows

**Comment by mcDandy** [comment]
[code]daniel@daniel-pc:~$ pip3 show -f hypergan
Name: hypergan
Version: 0.9.4
Summary: A customizable generative adversarial network with reproducible configurations.  Build your own content generator.
Home-page: https://github.com/255BITS/hypergan
Author: hypergan developers
Author-email: mikkel@255bits.com
License: MIT
Location: /home/daniel/.local/lib/python3.5/site-packages
Requires: 
Files:
  ../../../bin/hypergan
  hypergan-0.9.4.dist-info/DESCRIPTION.rst
  hypergan-0.9.4.dist-info/INSTALLER
  hypergan-0.9.4.dist-info/METADATA
  hypergan-0.9.4.dist-info/RECORD
  hypergan-0.9.4.dist-info/WHEEL
  hypergan-0.9.4.dist-info/metadata.json
  hypergan-0.9.4.dist-info/top_level.txt
  hypergan/__init__.py
  hypergan/__pycache__/__init__.cpython-35.pyc
  hypergan/__pycache__/cli.cpython-35.pyc
  hypergan/__pycache__/configuration.cpython-35.pyc
  hypergan/__pycache__/gan.cpython-35.pyc
  hypergan/__pycache__/gan_component.cpython-35.pyc
  hypergan/__pycache__/multi_component.cpython-35.pyc
  hypergan/__pycache__/pygame_viewer.cpython-35.pyc
  hypergan/__pycache__/viewer.cpython-35.pyc
  hypergan/cli.py
  hypergan/configuration.py
  hypergan/configurations/__init__.py
  hypergan/configurations/__pycache__/__init__.cpython-35.pyc
  hypergan/configurations/align.json
  hypergan/configurations/alpha-default-tiny.json
  hypergan/configurations/alpha-default.json
  hypergan/configurations/alphagan-lsgan.json
  hypergan/configurations/began-optimized.json
  hypergan/configurations/began.json
  hypergan/configurations/colorizer-default.json
  hypergan/configurations/cramer-gan.json
  hypergan/configurations/dcgan.json
  hypergan/configurations/default.json
  hypergan/configurations/improved-gan-resizable.json
  hypergan/configurations/improved-gan.json
  hypergan/configurations/improved-wgan.json
  hypergan/configurations/least-squares-gan-optimized.json
  hypergan/configurations/least-squares-gan.json
  hypergan/configurations/softmax-gan.json
  hypergan/configurations/wgan.json
  hypergan/discriminators/__init__.py
  hypergan/discriminators/__pycache__/__init__.cpython-35.pyc
  hypergan/discriminators/__pycache__/autoencoder_discriminator.cpython-35.pyc
  hypergan/discriminators/__pycache__/base_discriminator.cpython-35.pyc
  hypergan/discriminators/__pycache__/common.cpython-35.pyc
  hypergan/discriminators/__pycache__/cramer_discriminator.cpython-35.pyc
  hypergan/discriminators/__pycache__/dcgan_discriminator.cpython-35.pyc
  hypergan/discriminators/__pycache__/fully_connected_discriminator.cpython-35.pyc
  hypergan/discriminators/__pycache__/pyramid_discriminator.cpython-35.pyc
  hypergan/discriminators/autoencoder_discriminator.py
  hypergan/discriminators/base_discriminator.py
  hypergan/discriminators/common.py
  hypergan/discriminators/cramer_discriminator.py
  hypergan/discriminators/dcgan_discriminator.py
  hypergan/discriminators/fully_connected_discriminator.py
  hypergan/discriminators/pyramid_discriminator.py
  hypergan/encoders/__init__.py
  hypergan/encoders/__pycache__/__init__.cpython-35.pyc
  hypergan/encoders/__pycache__/base_encoder.cpython-35.pyc
  hypergan/encoders/__pycache__/category_encoder.cpython-35.pyc
  hypergan/encoders/__pycache__/uniform_encoder.cpython-35.pyc
  hypergan/encoders/base_encoder.py
  hypergan/encoders/category_encoder.py
  hypergan/encoders/uniform_encoder.py
  hypergan/gan.py
  hypergan/gan_component.py
  hypergan/gans/__init__.py
  hypergan/gans/__pycache__/__init__.cpython-35.pyc
  hypergan/gans/__pycache__/aligned_alpha_gan.cpython-35.pyc
  hypergan/gans/__pycache__/aligned_gan.cpython-35.pyc
  hypergan/gans/__pycache__/alpha_gan.cpython-35.pyc
  hypergan/gans/__pycache__/autoencoder_gan.cpython-35.pyc
  hypergan/gans/__pycache__/base_gan.cpython-35.pyc
  hypergan/gans/__pycache__/standard_gan.cpython-35.pyc
  hypergan/gans/aligned_alpha_gan.py
  hypergan/gans/aligned_gan.py
  hypergan/gans/alpha_gan.py
  hypergan/gans/autoencoder_gan.py
  hypergan/gans/base_gan.py
  hypergan/gans/standard_gan.py
  hypergan/generators/__init__.py
  hypergan/generators/__pycache__/__init__.cpython-35.pyc
  hypergan/generators/__pycache__/base_generator.cpython-35.pyc
  hypergan/generators/__pycache__/common.cpython-35.pyc
  hypergan/generators/__pycache__/dcgan_generator.cpython-35.pyc
  hypergan/generators/__pycache__/deconv_generator.cpython-35.pyc
  hypergan/generators/__pycache__/fully_connected_generator.cpython-35.pyc
  hypergan/generators/__pycache__/resize_conv_generator.cpython-35.pyc
  hypergan/generators/base_generator.py
  hypergan/generators/common.py
  hypergan/generators/dcgan_generator.py
  hypergan/generators/deconv_generator.py
  hypergan/generators/fully_connected_generator.py
  hypergan/generators/resize_conv_generator.py
  hypergan/inputs/__init__.py
  hypergan/inputs/__pycache__/__init__.cpython-35.pyc
  hypergan/inputs/__pycache__/audio_loader.cpython-35.pyc
  hypergan/inputs/__pycache__/image_loader.cpython-35.pyc
  hypergan/inputs/__pycache__/resize_audio_patch.cpython-35.pyc
  hypergan/inputs/__pycache__/resize_image_patch.cpython-35.pyc
  hypergan/inputs/audio_loader.py
  hypergan/inputs/image_loader.py
  hypergan/inputs/resize_audio_patch.py
  hypergan/inputs/resize_image_patch.py
  hypergan/losses/__init__.py
  hypergan/losses/__pycache__/__init__.cpython-35.pyc
  hypergan/losses/__pycache__/base_loss.cpython-35.pyc
  hypergan/losses/__pycache__/boundary_equilibrium_loss.cpython-35.pyc
  hypergan/losses/__pycache__/category_loss.cpython-35.pyc
  hypergan/losses/__pycache__/cramer_loss.cpython-35.pyc
  hypergan/losses/__pycache__/lamb_gan_loss.cpython-35.pyc
  hypergan/losses/__pycache__/least_squares_loss.cpython-35.pyc
  hypergan/losses/__pycache__/softmax_loss.cpython-35.pyc
  hypergan/losses/__pycache__/standard_loss.cpython-35.pyc
  hypergan/losses/__pycache__/supervised_loss.cpython-35.pyc
  hypergan/losses/__pycache__/wasserstein_loss.cpython-35.pyc
  hypergan/losses/base_loss.py
  hypergan/losses/boundary_equilibrium_loss.py
  hypergan/losses/category_loss.py
  hypergan/losses/cramer_loss.py
  hypergan/losses/lamb_gan_loss.py
  hypergan/losses/least_squares_loss.py
  hypergan/losses/softmax_loss.py
  hypergan/losses/standard_loss.py
  hypergan/losses/supervised_loss.py
  hypergan/losses/wasserstein_loss.py
  hypergan/multi_component.py
  hypergan/ops/__init__.py
  hypergan/ops/__pycache__/__init__.cpython-35.pyc
  hypergan/ops/tensorflow/__init__.py
  hypergan/ops/tensorflow/__pycache__/__init__.cpython-35.pyc
  hypergan/ops/tensorflow/__pycache__/activations.cpython-35.pyc
  hypergan/ops/tensorflow/__pycache__/extended_ops.cpython-35.pyc
  hypergan/ops/tensorflow/__pycache__/layer_regularizers.cpython-35.pyc
  hypergan/ops/tensorflow/__pycache__/ops.cpython-35.pyc
  hypergan/ops/tensorflow/activations.py
  hypergan/ops/tensorflow/extended_ops.py
  hypergan/ops/tensorflow/layer_regularizers.py
  hypergan/ops/tensorflow/ops.py
  hypergan/pygame_viewer.py
  hypergan/samplers/__init__.py
  hypergan/samplers/__pycache__/__init__.cpython-35.pyc
  hypergan/samplers/__pycache__/aligned_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/alphagan_random_walk_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/autoencode_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/base_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/batch_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/began_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/grid_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/random_walk_sampler.cpython-35.pyc
  hypergan/samplers/__pycache__/static_batch_sampler.cpython-35.pyc
  hypergan/samplers/aligned_sampler.py
  hypergan/samplers/alphagan_random_walk_sampler.py
  hypergan/samplers/autoencode_sampler.py
  hypergan/samplers/base_sampler.py
  hypergan/samplers/batch_sampler.py
  hypergan/samplers/began_sampler.py
  hypergan/samplers/grid_sampler.py
  hypergan/samplers/random_walk_sampler.py
  hypergan/samplers/static_batch_sampler.py
  hypergan/search/__init__.py
  hypergan/search/__pycache__/__init__.cpython-35.pyc
  hypergan/search/__pycache__/aligned_random_search.cpython-35.pyc
  hypergan/search/__pycache__/alphagan_random_search.cpython-35.pyc
  hypergan/search/__pycache__/default_configurations.cpython-35.pyc
  hypergan/search/__pycache__/random_search.cpython-35.pyc
  hypergan/search/aligned_random_search.py
  hypergan/search/alphagan_random_search.py
  hypergan/search/default_configurations.py
  hypergan/search/random_search.py
  hypergan/trainers/__init__.py
  hypergan/trainers/__pycache__/__init__.cpython-35.pyc
  hypergan/trainers/__pycache__/alternating_trainer.cpython-35.pyc
  hypergan/trainers/__pycache__/base_trainer.cpython-35.pyc
  hypergan/trainers/__pycache__/multi_step_trainer.cpython-35.pyc
  hypergan/trainers/__pycache__/proportional_control_trainer.cpython-35.pyc
  hypergan/trainers/alternating_trainer.py
  hypergan/trainers/base_trainer.py
  hypergan/trainers/multi_step_trainer.py
  hypergan/trainers/proportional_control_trainer.py
  hypergan/viewer.py
[\code]

**Comment by mcDandy** [comment]
I misscounted  the number of [floder ups] and thought that the "hypergan" file is in completly diferrent place. It's in /home/daniel/.local/bin not /bin that I thought

**Comment by martyn** [comment]
Great.  I hope it's working now :)

**Comment by mcDandy** [comment]
Tried to use virtualenv command but it throws an error


daniel@daniel-pc:~$ virtualenv --system-site-packages -p python3 hypergan
Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/daniel/hypergan/bin/python3
Not overwriting existing python script /home/daniel/hypergan/bin/python (you must use /home/daniel/hypergan/bin/python3)
Installing setuptools, pkg_resources, pip, wheel...
  Complete output from command /home/daniel/hypergan/bin/python3 - setuptools pkg_resources pip wheel:
  Collecting setuptools
  Downloading setuptools-36.2.0-py2.py3-none-any.whl (477kB)
Collecting pkg_resources
Collecting pip
Collecting wheel
Installing collected packages: setuptools, pkg-resources, pip, wheel
Exception:
Traceback (most recent call last):
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/commands/install.py", line 360, in run
    prefix=options.prefix_path,
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/usr/share/python-wheels/pip-9.0.1-py2.py3-none-any.whl/pip/wheel.py", line 323, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/home/daniel/hypergan/lib/python3.5/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
PermissionError: [Errno 13] Operace zamítnuta: '/home/daniel/hypergan/lib/python3.5/site-packages/pkg_resources-0.0.0.dist-info/RECORD'
----------------------------------------
...Installing setuptools, pkg_resources, pip, wheel...done.
Traceback (most recent call last):
  File "/usr/bin/virtualenv", line 11, in <module>
    load_entry_point('virtualenv==15.1.0', 'console_scripts', 'virtualenv')()
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 724, in main
    symlink=options.symlink)
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 992, in create_environment
    download=download,
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 922, in install_wheel
    call_subprocess(cmd, show_stdout=False, extra_env=env, stdin=SCRIPT)
  File "/usr/lib/python3/dist-packages/virtualenv.py", line 817, in call_subprocess
    % (cmd_desc, proc.returncode))
OSError: Command /home/daniel/hypergan/bin/python3 - setuptools pkg_resources pip wheel failed with error code 2

**Comment by mcDandy** [comment]
still not working

**Comment by falkartis** [comment]
may it make a difference whether you install it using sudo or as normal user, in case of sudo pip tells me to use sudo's -H argument.

```
sudo -H pip3 install hypergan --upgrade
sudo -H pip3 install numpy tensorflow-gpu hyperchamber pillow pygame
```

this worked for me, but now I have to open another issue because of another error XD, I don't surrender XD.

**Comment by ThatBulgarian** [comment]
I've got this same problem in windows... the hypergan file (and the bin folder that it's meant to be in) arent there

**Comment by martyn** [comment]
This should be fixed in the upcoming 1.0 release since installation has been simplified. Also we're planning to release a windows installer eventually.

**Comment by MasterCubo** [comment]
Update: Still does not work in 1.0
